packages/reactive-element/src/test/std-decorators/property_test.ts

```diff
@@ -0,0 +1,546 @@
+
+      @property({
+        attribute: 'all-attr',
+        hasChanged,
+        converter: {fromAttribute, toAttribute},
+        reflect: true,
+      })
+      accessor all = 10;
+
+      updateCount = 0;
+
+      static override get properties() {
+        return {
+          noAttr: {attribute: false},
+          atTr: {attribute: true},
+          customAttr: {attribute: 'custom', reflect: true},
+        };
+      }
+
+      declare noAttr: string | undefined;
+      declare atTr: string | undefined;
+      declare customAttr: string | undefined;
+
+      constructor() {
+        super();
+        this.noAttr = 'noAttr';
+        this.atTr = 'attr';
+        this.customAttr = 'customAttr';
+      }
+
+      override update(changed: PropertyValues) {
+        this.updateCount++;
+        super.update(changed);
+      }
+    }
+    customElements.define(generateElementName(), E);
+    const el = new E();
+    container.appendChild(el);
+    await el.updateComplete;
+    assert.equal(el.updateCount, 1);
+    assert.equal(el.noAttr, 'noAttr');
+    assert.equal(el.atTr, 'attr');
+    assert.equal(el.customAttr, 'customAttr');
+    assert.equal(el.hasChanged, 10);
+    assert.equal(el.fromAttribute, 1);
+    assert.equal(el.toAttribute, 1);
+    assert.equal(el.getAttribute('toattribute'), null);
+    assert.equal(el.all, 10);
+    assert.equal(el.getAttribute('all-attr'), null);
+    el.setAttribute('noattr', 'noAttr2');
+    el.setAttribute('attr', 'attr2');
+    el.setAttribute('custom', 'customAttr2');
+    el.setAttribute('fromattribute', '2attr');
+    el.toAttribute = 2;
+    el.all = 5;
+    await el.updateComplete;
+    assert.equal(el.updateCount, 2);
+    assert.equal(el.noAttr, 'noAttr');
+    assert.equal(el.atTr, 'attr2');
+    assert.equal(el.customAttr, 'customAttr2');
+    assert.equal(el.fromAttribute, 2);
+    assert.equal(el.toAttribute, 2);
+    assert.equal(el.getAttribute('toattribute'), '2-attr');
+    assert.equal(el.all, 5);
+    el.all = 15;
+    await el.updateComplete;
+    assert.equal(el.updateCount, 3);
+    assert.equal(el.all, 15);
+    assert.equal(el.getAttribute('all-attr'), '15-attr');
+    el.setAttribute('all-attr', '16-attr');
+    await el.updateComplete;
+    assert.equal(el.updateCount, 4);
+    assert.equal(el.getAttribute('all-attr'), '16-attr');
+    assert.equal(el.all, 16);
+    el.hasChanged = 5;
+    await el.updateComplete;
+    assert.equal(el.hasChanged, 5);
+    assert.equal(el.updateCount, 4);
+    el.hasChanged = 15;
+    await el.updateComplete;
+    assert.equal(el.hasChanged, 15);
+    assert.equal(el.updateCount, 5);
+    el.setAttribute('all-attr', '5-attr');
+    await el.updateComplete;
+    assert.equal(el.all, 5);
+    assert.equal(el.updateCount, 5);
+    el.all = 15;
+    await el.updateComplete;
+    assert.equal(el.all, 15);
+    assert.equal(el.updateCount, 6);
+  });
+
+  // This test isn't valid. You can not override a field that's initialized
```

Comment:
This is a limitation is JS now. You can't do:

```ts
class A {
  constructor() {
    this.foo = 1;
  }
}
class B extends A {
  accessor foo;
}
```

because A's `this.foo = 1` runs before B's foo accessor has it's private storage defined. That's what this test is trying to do, so at least that bit is not even testable. The main idea that a subclass can't interfere with a superclass it good though... I'm rewriting to avoid the timing problem.